### PR TITLE
TINKERPOP-2749: Gremlin Javascript, Gremlint Windows build error fixes

### PR DESCRIPTION
**Pull request by L0Lmaker**
Currently, Tinkerpop does not build on Windows. Running `mvn clean install` causes errors in `gremlin-javascript` and other modules.

This PR fixes the errors caused by `gremlin-javascript`. It also adds a Windows build check to the GitHub Actions workflows.

Issues addressed:
1. Fixed an issue where groovy script defined in `gremlin-javascript/pom.xml` receives an incorrectly formatted path for Windows. Single `\` characters are invalid in groovy and using `${project.basedir}` returns a string with non-escaped `\` characters.
2. Fixed an issue where lint/line-break checks would fail with message: `Expected linebreaks to be 'LF' but found 'CRLF'`
3. Fixed an issue where relative addressing is used, which was causing `'.' is not recognized as an internal or external command, operable program or batch file.` errors.
4. Fixed an issue where incorrectly formatted paths taken from `pom.xml` properties would lead to `test-server-start.groovy` initializing integration test server incorrectly on Windows.

Link to Issue: [TINKERPOP-2749](https://issues.apache.org/jira/browse/TINKERPOP-2749)

### CODE REVIEW

**Comment by lyndonbauto** [comment]
@L0Lmaker Looks like we got a conflict now, could you rebase?

File: gremlin-javascript/src/main/javascript/gremlin-javascript/.eslintrc.js [hunk 1]
```diff
@@ -31,9 +31,9 @@ module.exports = {
   extends: ['eslint:recommended', 'prettier'],
   plugins: ['prettier'],
   rules: {
-    'prettier/prettier': ['error'],
```
Comment (FlorianHockmann) [inline gremlin-javascript/src/main/javascript/gremlin-javascript/.eslintrc.js:34]: The changes in this file allow line endings to be `CRLF`, right? Is that really necessary? I guess we'd prefer if only Unix style line endings were committed, but maybe I'm missing something here?

**Review by FlorianHockmann — COMMENTED** [review]
Great work, I just executed the tests locally on my Windows machine and they passed! It's really nice to get better support for developing on Windows :tada:
Also good to have this tested in CI so we can be sure that we don't accidentally break Windows support in the future.

I've only added one question, but these changes look already good to me (a rebase is of course still necessary).

VOTE +1

File: gremlin-javascript/src/main/javascript/gremlin-javascript/.eslintrc.js (hunk 1 above)
Comment (L0Lmaker) [reply gremlin-javascript/src/main/javascript/gremlin-javascript/.eslintrc.js:34]: There are two ways we could resolve this. Either allow for `CRLF` formatted files or convert the existing javascript files that use the `CRLF` format to `LF`.  The former felt like a simpler fix which is why I disabled the check for line separators.

If we want to keep it only Unix style, we could go ahead with converting all the JS files that are affected.

**Comment by lyndonbauto** [comment]
I think we are okay to move forward with this once the actions pass. I will handle merging this through.

File: gremlin-javascript/src/main/javascript/gremlin-javascript/.eslintrc.js (hunk 1 above)
Comment (lyndonbauto) [reply gremlin-javascript/src/main/javascript/gremlin-javascript/.eslintrc.js:34]: I think we can leave this one just to move this along. If someone wants to go through and convert all the line endings, that can be done separately.

Comment (FlorianHockmann) [reply gremlin-javascript/src/main/javascript/gremlin-javascript/.eslintrc.js:34]: Oh, I wasn't aware that there are already files with `CRLF`. Then I understand the change. Thanks for the explanation!

**Pull request merged**